### PR TITLE
[incubator/elasticsearch] rolling update strategy for data nodes

### PR DESCRIPTION
**Pull request by kimxogus**
Signed-off-by: Taehyun Kim <kgyoo8232@gmail.com>

**What this PR does / why we need it**:
As data nodes have pre-stop process which makes the node empty, rolling update on statefulset update will not make cluster downtime or other index related problems. I applied rolling update strategy to company monitoring cluster which stores more than 10gb of k8s cluster logs, and it seems fine.
I didn't applied rolling update strategy to master nodes because it causes cluster downtime.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by desaintmartin** [comment]
Could this be an option?
Could it be possible to have it for Master nodes?

File: incubator/elasticsearch/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: elasticsearch
 home: https://www.elastic.co/products/elasticsearch
-version: 1.6.3
+version: 1.6.4
```
Comment (cpanato) [inline incubator/elasticsearch/Chart.yaml]: should be `1.7.0` adding new functionality

Comment (kimxogus) [reply incubator/elasticsearch/Chart.yaml]: done :)

**Comment by desaintmartin** [comment]
Thanks! Shouldn't the default be the same for both (and maybe as a suggestion only, keep default behavior for StatefulSet v1beta1 which is OnDelete?)

**Pull request closed without merging**